IPFS Updates: 1.6

enoldev opened the pull request:
I've come up with the following subgoals:
* **1.61 –** Understand what is UnixFS and what problem it solves.
* **1.62 –** Understand how files with more than a single block are labeled.
* **1.63 (ONLY DEEP RESIDENTS)–** Understand the protobuf format, and its main fields.

Things to note:
- I haven't included chunking and layouts because it's necessary to talk about IPLD stuff to explain it, and we haven't introduced IPLD at this point. My suggestion is to add a "UnixFS & IPLD" section in the IPLD curriculum
- I've included content only for deep people in this section. Note the `{{% level type="[deep]" %}}` tag

mrodriguez3313 commented:
Running this on my machine, I tried both Edge and Chrome, neither hid the content when switching between Shallow and Deep. 

What I tried instead was I changed the short code to use Divs instead of spans (how you originally had it) and it worked.

That being said idk if we need to have the `level` shortcodes in this UnixFS lesson since the whole page itself is a deep page. And with the new [react components you created](https://github.com/protocol/launchpad/pull/302), it will be much harder for a shallow person to find the content. AND idk if a deep user will know to change to Shallow for this one page to just go over a smaller portion of the curriculum.

What do you think?

enoldev commented:
About the hiding feature:
- It was failing to hide the deep objective because the Markdown needed a break line. Fixed, it should work now
- That's the reason why I changed the div to span, to avoid the unnecessary whitespace

About having a shallow/deep content
- I forgot to add this file for both shallow and deep. Fixed
- I tried to explain the first part of this section for shallow people (that's why used several diagrams why files examples). I think it's nice for shallow people to know, at least, that there is a "system" to label blocks. Otherwise it's going to be more difficult for them to understand IPLD

mrodriguez3313 commented:
> I forgot to add this file for both shallow and deep. Fixed

Ahh okay awesome. Thank makes sense now.

> it's still showing that weird whitespace, I gotta find a way to avoid it.

A bad way to do it would be to add `margin-bottom` CSS property to all `<li>` and remove the margins from all `<ul>`.  checkout this solution. https://github.com/protocol/launchpad/tree/MAR/ipfs1.6

mrodriguez3313 commented on content/en/curriculum/ipfs/unixfs/index.md:
```diff
@@ -0,0 +1,86 @@
+
+## Learning Objectives
+
+**IPFS 1.4 Be able to explain the basics of UnixFS**
```
```suggestion
**IPFS 1.6 Be able to explain the basics of UnixFS**
```

enoldev commented:
@mrodriguez3313 I'm working in a PR to add objectives in a simpler way. I'll open it and we can make design changes there
